test/IntegrationTests/MSTest.IntegrationTests/OutputTests.cs

```diff
@@ -16,17 +18,56 @@ public class OutputTests : CLITestBase
 {
     private const string TestAssetName = "OutputTestProject";
 
+    private const string RunSettingXml =
+        @"<RunSettings>   
+                <RunConfiguration>  
+                    <DisableAppDomain>False</DisableAppDomain>   
+                </RunConfiguration>  
+            </RunSettings>";
+
     public void OutputIsNotMixedWhenTestsRunInParallel()
     {
-        ValidateOutputForClass("UnitTest1");
+        ValidateOutputForClass("UnitTest1", RunSettingXml);
     }
 
     public void OutputIsNotMixedWhenAsyncTestsRunInParallel()
     {
-        ValidateOutputForClass("UnitTest2");
+        ValidateOutputForClass("UnitTest2", RunSettingXml);
+    }
+
+    public void TestContextWriteLine_WhenTestsRunInParallel_OutputIsNotMixed()
+    {
+        // By default no appDomain.
+        ValidateTestContextOutputForClass("UnitTest3");
+    }
+
+    private void ValidateTestContextOutputForClass(string className)
+    {
+        // Arrange
+        var assemblyPath = GetAssetFullPath(TestAssetName);
+
+        // Act
+        var testCases = DiscoverTests(assemblyPath, null, RunSettingXml).Where(tc => tc.FullyQualifiedName.Contains(className)).ToList();
+        testCases.Should().HaveCount(4);
+        testCases.Should().NotContainNulls();
+
+        var testResults = RunTests(testCases);
+        testResults.Should().HaveCount(4);
+        testResults.Should().NotContainNulls();
+
+        // Assert
+        // Ensure that some tests are running in parallel, because otherwise the output just works correctly.
+        var firstEnd = testResults.Min(t => t.EndTime);
+        var someStartedBeforeFirstEnded = testResults.Where(t => t.EndTime != firstEnd).Any(t => firstEnd > t.StartTime);
+        someStartedBeforeFirstEnded.Should().BeTrue("Tests must run in parallel, but there were no other tests that started, before the first one ended.");
+
+        ValidateOutputIsNotMixed(testResults, "TestMethod2(DataRow1)", new[] { "TestMethod1", "TestMethod3" }, IsStandardOutputMessage, "TestMethod2 (DataRow1)");
```

Comment:
I think you have a space (don't ask me why) between the method name and the parameter types (e.g. `MyMethod (string)`).